core/domain/stats_services_test.py

```diff
@@ -1151,6 +1159,196 @@ def test_cst_exp_issue_is_updated_when_state_is_renamed(self):
+
+        with self.assertRaisesRegex(
+            NotImplementedError,
+            'method is missing from the derived class.'):
+            stats_services.get_playthrough_from_model(model)
+            stats_models.CURRENT_ACTION_SCHEMA_VERSION = 1
+
+    def test_get_exp_issues_from_model(self):
+        stats_services.save_exp_issues_model(
+            stats_domain.ExplorationIssues(self.exp.id, self.exp.version, [
+                self._create_cst_exp_issue(
+                    [self._create_cst_playthrough(['A', 'B', 'A'])],
+                    ['A', 'B', 'A'])
+            ]))
+
+        exp_services.update_exploration(self.owner_id, self.exp.id, [
+            exp_domain.ExplorationChange({
+                'cmd': 'rename_state',
+                'old_state_name': 'A',
+                'new_state_name': 'Z',
+            })
+        ], 'change')
+
+        exp_issues_model = stats_models.ExplorationIssuesModel.get_model(
+            self.exp.id, self.exp.version)
+
+        stats_models.CURRENT_ISSUE_SCHEMA_VERSION = 0
+
+        with self.assertRaisesRegex(
+            Exception, 'Sorry, we can only process '
+            'v1-v%d and unversioned issue schemas '
+            'at present.' %
+            stats_models.CURRENT_ISSUE_SCHEMA_VERSION):
+            stats_services.get_exp_issues_from_model(exp_issues_model)
+            stats_models.CURRENT_ISSUE_SCHEMA_VERSION = 1
+
+    def test_while_get_exp_issues_from_model(self):
+        stats_models.CURRENT_ISSUE_SCHEMA_VERSION = 1
+        stats_services.save_exp_issues_model(
+            stats_domain.ExplorationIssues(self.exp.id, self.exp.version, [
+                self._create_cst_exp_issue(
+                    [self._create_cst_playthrough(['A', 'B', 'A'])],
+                    ['A', 'B', 'A'])
+            ]))
+
+        exp_services.update_exploration(self.owner_id, self.exp.id, [
+            exp_domain.ExplorationChange({
+                'cmd': 'rename_state',
+                'old_state_name': 'A',
+                'new_state_name': 'Z',
+            })
+        ], 'change')
+
+        exp_issues_model = stats_models.ExplorationIssuesModel.get_model(
+            self.exp.id, self.exp.version)
+
+        stats_models.CURRENT_ISSUE_SCHEMA_VERSION = 2
+
+        with self.assertRaisesRegex(
+            NotImplementedError,
+            'method is missing from the derived class.'):
+            stats_services.get_exp_issues_from_model(exp_issues_model)
+            stats_models.CURRENT_ISSUE_SCHEMA_VERSION = 1
+
+    def test_assign_playthrough_to_corresponding_issue(self):
+        stats_services.save_exp_issues_model(
+            stats_domain.ExplorationIssues(self.exp.id, self.exp.version, [
+                self._create_cst_exp_issue(
+                    [self._create_cst_playthrough(['A', 'B', 'A'])],
+                    ['A', 'B', 'A'])
+            ]))
+
+        exp_services.update_exploration(self.owner_id, self.exp.id, [
```

Comment:
I don't really understand why you need this stuff if you're just testing assignation of a playthrough to an issue? Try to keep the test minimal.